src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -717,9 +726,78 @@ address StubGenerator::generate_disjoint_copy_avx3_masked(address* entry, const
   __ leave(); // required for proper stackwalking of RuntimeStub frame
   __ ret(0);
 
+  if (MaxVectorSize == 64) {
+    __ BIND(L_copy_large);
+    arraycopy_avx3_large(to, from, temp1, temp2, temp3, temp4, count, xmm1, xmm2, xmm3, xmm4, shift);
+    __ jmp(L_finish);
+  }
   return start;
 }
 
+void StubGenerator::arraycopy_avx3_large(Register to, Register from, Register temp1, Register temp2,
+                                         Register temp3, Register temp4, Register count,
+                                         XMMRegister xmm1, XMMRegister xmm2, XMMRegister xmm3,
+                                         XMMRegister xmm4, int shift) {
+
+  // Type(shift)           byte(0), short(1), int(2),   long(3)
+  int loop_size[]        = { 256,     128,       64,      32};
+  int threshold[]        = { 4096,    2048,     1024,    512};
+
+  Label L_main_loop_large;
+  Label L_tail_large;
+  Label L_exit_large;
+  Label L_entry_large;
+  Label L_main_pre_loop_large;
+  Label L_pre_main_post_large;
+
+  if (MaxVectorSize == 64) {
+    __ BIND(L_entry_large);
+
+    __ BIND(L_pre_main_post_large);
+    // Partial copy to make dst address 64 byte aligned.
+    __ movq(temp2, to);
+    __ andq(temp2, 63);
+    __ jcc(Assembler::equal, L_main_pre_loop_large);
+
+    __ negptr(temp2);
+    __ addq(temp2, 64);
+    if (shift) {
+      __ shrq(temp2, shift);
+    }
+    __ movq(temp3, temp2);
+    copy64_masked_avx(to, from, xmm1, k2, temp3, temp4, temp1, shift, 0);
```

Comment:
The last argument should be "true" or "1" instead of "0" or "false".  This is as temp3 (length) could be less than 32 as well. This case is only handled when use64byteVector argument is true.

Reply:
Thanks, done.